### PR TITLE
Use coursier.Versions in checkForUpdate

### PR DESCRIPTION
**Pull request by fthomas**
This contains @manuelcueto's changes from #1131 targeted to master.

We can use this PR for further discussion and changes.

### CODE REVIEW

**Comment by manuelcueto** [comment]
It looks good, removes a  lot of code that was a bit hard to follow and wasn't tested. my first commit was just to use it but i thought that it was missing some logic that sbtAlg was doing so that's why i only created a draft. but it seems you got it to a much better shape. And the bump in performance is very exciting. 
My only question is why is this only used for `pruneRepos = true` only?

**Comment by fthomas** [comment]
`checkForUpdates` is only used in the implementation of the repos pruning scheme as described in #707. It is not (yet) a full replacement for using `sbt-updates` since it doesn't use any custom resolvers and credentials that might be defined in the sbt build and it also ignores if an update is available for all `crossScalaVersions`. If we could implement these features and replace sbt-updates, Scala Steward could be blazingly fast.

**Comment by fthomas** [comment]
I'm merging this now because I'm curious how this reduces the runtime of my public Scala Steward instance. I'll report back once the numbers are in. :timer_clock:

Thanks again @manuelcueto!

**Comment by fthomas** [comment]
The numbers are in. Instead of 80 minutes, `checkForUpdates` now takes only 8 minutes! :confetti_ball:

**Comment by manuelcueto** [comment]
😍
